mautic/mautic
# Fix embedding images in mails

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #13202 



#### Description:

In https://github.com/mautic/mautic/pull/11613 Mautic switched to Symfony Mailer.
This required some methods in e.g. `MailHelper`  to be adapted to the new logic.

the logic in `MailHelper::convertEmbedImages` was not changed in a compatible way, as `Mail::embedFromPath` doesn't return the same data as the original `Swift_Message:embed`.

This PR addresses this by changing the logic.

The open question is if it is ok to fetch any image hosted outside of the Mautic instance, as I have some security concerns there. Looking at the original code, it was possible to do so.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. set the `'mailer_convert_embed_images' => true,` in the `config.php` file
3. send an email to a contact, and add an image to the email body.
4. verify the source code of the email contains `cid:` references in the email body, and the referenced image as attachment in the email.
